### PR TITLE
CodeFixService simplifications

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
I'm working in this area, and makign a bunch of a changes.  This extracts out some general cleanup that comes with no behavior changes.

### CODE REVIEW

File: src/Features/Core/Portable/CodeFixes/CodeFixService.cs [hunk 1]
```diff
@@ -39,25 +39,23 @@ internal partial class CodeFixService : ICodeFixService
             new((d1, d2) => DiagnosticId.CompareOrdinal(d1.Id, d2.Id));
 
         private readonly IDiagnosticAnalyzerService _diagnosticService;
+        private readonly ImmutableArray<Lazy<CodeFixProvider, CodeChangeProviderMetadata>> _fixers;
+        private readonly Dictionary<string, List<Lazy<CodeFixProvider, CodeChangeProviderMetadata>>> _fixersPerLanguageMap;
 
-        private readonly Func<Workspace, ImmutableDictionary<LanguageKind, Lazy<ImmutableDictionary<DiagnosticId, ImmutableArray<CodeFixProvider>>>>> _getWorkspaceFixersMap;
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/CodeFixes/CodeFixService.cs:43]: remove unnecessary callbacks.  These can just be normal methods in this type.

File: src/Features/Core/Portable/CodeFixes/CodeFixService.cs [hunk 2]
```diff
@@ -39,25 +39,23 @@ internal partial class CodeFixService : ICodeFixService
             new((d1, d2) => DiagnosticId.CompareOrdinal(d1.Id, d2.Id));
 
         private readonly IDiagnosticAnalyzerService _diagnosticService;
+        private readonly ImmutableArray<Lazy<CodeFixProvider, CodeChangeProviderMetadata>> _fixers;
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/CodeFixes/CodeFixService.cs:42]: don't store as an IEnumerable.

File: src/Features/Core/Portable/CodeFixes/CodeFixService.cs [hunk 3]
```diff
@@ -39,25 +39,23 @@ internal partial class CodeFixService : ICodeFixService
...
 
-        private readonly ConditionalWeakTable<AnalyzerReference, ProjectCodeFixProvider> _analyzerReferenceToFixersMap;
-        private readonly ConditionalWeakTable<AnalyzerReference, ProjectCodeFixProvider>.CreateValueCallback _createProjectCodeFixProvider;
-
-        private readonly ImmutableDictionary<LanguageKind, Lazy<ImmutableArray<IConfigurationFixProvider>>> _configurationProvidersMap;
-        private readonly IEnumerable<Lazy<IErrorLoggerService>> _errorLoggers;
-
-        private ImmutableDictionary<object, FixAllProviderInfo?> _fixAllProviderMap;
+        private ImmutableDictionary<CodeFixProvider, ImmutableArray<DiagnosticId>> _fixerToFixableIdsMap = ImmutableDictionary<CodeFixProvider, ImmutableArray<DiagnosticId>>.Empty;
+        private ImmutableDictionary<object, FixAllProviderInfo?> _fixAllProviderMap = ImmutableDictionary<object, FixAllProviderInfo?>.Empty;
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/CodeFixes/CodeFixService.cs:58]: move mutable fields to bottom.

File: src/Features/Core/Portable/CodeFixes/CodeFixService.cs [hunk 4]
```diff
@@ -67,24 +65,15 @@ internal partial class CodeFixService : ICodeFixService
...
-            var configurationProvidersPerLanguageMap = configurationProviders.ToPerLanguageMapWithMultipleLanguages();
-
-            _getWorkspaceFixersMap = workspace => GetFixerPerLanguageMap(fixersPerLanguageMap, workspace);
-            _configurationProvidersMap = GetConfigurationProvidersPerLanguageMap(configurationProvidersPerLanguageMap);
+            _fixers = fixers.ToImmutableArray();
+            _fixersPerLanguageMap = _fixers.ToPerLanguageMapWithMultipleLanguages();
 
-            // REVIEW: currently, fixer's priority is statically defined by the fixer itself. might considering making it more dynamic or configurable.
-            _getFixerPriorityMap = workspace => GetFixerPriorityPerLanguageMap(fixersPerLanguageMap, workspace);
+            _lazyFixerToMetadataMap = new(() => fixers.Where(service => service.IsValueCreated).ToImmutableDictionary(service => service.Value, service => service.Metadata));
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/CodeFixes/CodeFixService.cs:74]: Out of curiosity, when this runs in practice are there any fixers already created or is this just initializing an empty map?

Comment (jasonmalinowski) [reply src/Features/Core/Portable/CodeFixes/CodeFixService.cs:74]: (why this is stuffed in a Lazy of its own is terrifying...)

**Review by jasonmalinowski — APPROVED** [review]
My only real complaint is there's still code left after all this cleanup. 😛 :shipit:

File: src/Features/Core/Portable/CodeFixes/CodeFixService.cs (hunk 4 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/CodeFixes/CodeFixService.cs:74]: @mavasani this is a bit scary.  this lazy is exposed through:

```c#
private ImmutableDictionary<CodeFixProvider, CodeChangeProviderMetadata> FixerToMetadataMap => _lazyFixerToMetadataMap.Value;
```

This means that we capture whatever the state of hte world is when this is computed.  Meaning if a service hasn't been created, then it's never in this map.  This seems fishy, esp in a multi-language world.  How do we know that we won't capture a snapshot of things in a bad state?

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/CodeFixes/CodeFixService.cs:74]: attempting to fix that here: https://github.com/dotnet/roslyn/pull/59586

**Pull request merged**